spacy/pipeline/entity_linker.py

```diff
@@ -220,11 +218,13 @@ def update(
             return losses
         validate_examples(examples, "EntityLinker.update")
         sentence_docs = []
-        docs = [eg.predicted for eg in examples]
-        if set_annotations:
-            # This seems simpler than other ways to get that exact output -- but
-            # it does run the model twice :(
-            predictions = self.model.predict(docs)
+        docs = []
+        for eg in examples:
+            eg.predicted.ents = eg.reference.ents
```

Comment:
Now that we're using `set_annotations` during the `update` flow, is it possible to learn from the predicted entities? We might still want to use the gold-standard entities as well as candidates, but we can at least learn with the predicted outputs?

Reply:
I'm not sure how it would work to learn from the predicted entities? We only have KB IDs for the gold-standard entities, and so we'd only be able to learn from those predicted ones that match gold-standard ones. 

I guess the only advantage might be that you'd focus the EL on those entities that the NER can actually predict, but at the same time a drop in recall for the NER will create a gap in knowledge for the EL?

Reply:
I think having incorrect entities can be important when figuring out thresholds, because you want to return `NIL` for them. Anyway, a question for another PR :).